### DISCUSSION TITLE
new monitor command for resetting the probe

### DISCUSSION THREAD

**Discussion by MikeDK** [blackmagic-debug/blackmagic#1239] [Ideas]
When the blackmagic probe is used in a CI environment and administered from home office without physical access, it would be very helpful to have a method to remotely restart the probe itself.... especially when using e.g. an stlink as probe where a FW update is only possible when in bootloader mode, where to get only after a usb replug at the moment.

I recently wrote a small patch to accomplish this and posted it in an issue related discussion.
But I really believe that this belongs into main branch as it does not hurt to have, but gives so much when needed ;)

```
diff --git a/src/command.c b/src/command.c
index 2758d31..c67bb18 100644
--- a/src/command.c
+++ b/src/command.c
@@ -46,6 +46,7 @@ struct command_s {
 
 static bool cmd_version(target *t, int argc, char **argv);
 static bool cmd_help(target *t, int argc, char **argv);
+static bool cmd_platform_boot(target *t, int argc, char **argv);
 
 static bool cmd_jtag_scan(target *t, int argc, char **argv);
 static bool cmd_swdp_scan(target *t, int argc, char **argv);
@@ -68,6 +69,7 @@ static bool cmd_debug_bmp(target *t, int argc, const char **argv);
 const struct command_s cmd_list[] = {
        {"version", (cmd_handler)cmd_version, "Display firmware version info"},
        {"help", (cmd_handler)cmd_help, "Display help for monitor commands"},
+       {"platform_boot", (cmd_handler)cmd_platform_boot, "Request reboot of probe"},
        {"jtag_scan", (cmd_handler)cmd_jtag_scan, "Scan JTAG chain for devices" },
        {"swdp_scan", (cmd_handler)cmd_swdp_scan, "Scan SW-DP for devices" },
        {"targets", (cmd_handler)cmd_targets, "Display list of available targets" },
@@ -167,6 +169,18 @@ bool cmd_help(target *t, int argc, char **argv)
        return true;
 }
 
+bool cmd_platform_boot(target *t, int argc, char **argv)
+{
+       (void)t;
+       (void)argc;
+       (void)argv;
+
+       gdb_out("Rebooting...");
+       platform_request_boot();
+
+       return true;
+}
+
 static bool cmd_jtag_scan(target *t, int argc, char **argv)
 {
        (void)t;
```

any opinions to the contrary? ;)

**Comment by UweBonnes** [comment]
When the probe is reactive, restart can be done by  USBvia DFU. Do you have no USB access to the BMD in your setup?

.

**Reply by dragonmux** [reply]
Actually, to be fair to them that is not true as while you can reboot into the bootloader with `dfu-util`'s `-e` option, you cannot re-reboot back into run mode this way. Only option there is power cycling, or.. this.. - a bug in the bootloader we intend to fix in the next release cycle

**Reply by Qyriad** [reply]
@dragonmux it actually is possible to request the BMP to leave DFU mode and reboot without flashing. iirc `dfu-util -s :leave` should do it (bmputil can also do it ofc).

(Edit: didn't see Uwe's already mentioned it in another thread. My bad!)

**Reply by dragonmux** [reply]
Aye, still want to address that `-e` doesn't work on the bootloader though which is.. problematic.. as that's supposed to issue a DFU_DETACH request and immediately pop back into run mode regardless of what extras (DFUse) are implemented. Hopefully though knowing those two dfu-util invocations can do the trick helps @MikeDK

**Comment by UweBonnes** [comment]
dfu-util -e : Go to bootloader
 dfu-util -s:leave Go from bootloader to application.

**Reply by dragonmux** [reply]
The bootloader should also implement `-e` - not doing so is a DFU base spec violation. Good to know about `-s:leave` though.
